### PR TITLE
Add new style for Style/MethodDefParentheses

### LINKED ISSUE #1729: Style/MethodDefParentheses doesn't support mix
OK. I came across an actual conflict. "Seattle-Style" eschews parens on method declarations unless the args span multiple lines:

```ruby
  def self.pie_chart(diameter, pct_green,
                     good_color = PNG::Color::Green,
                     bad_color = PNG::Color::Red)
```

This is usually only when there are long names or defaults that cause the declaration to go over 80 chars.

The defaults are currently:

```yaml
Style/MethodDefParentheses:
  EnforcedStyle: require_parentheses
  SupportedStyles:
    - require_parentheses
    - require_no_parentheses
```

but could use another, called `require_no_parethesis_except_multiline` or something.

### PR DESCRIPTION
**Pull request by alexdowad**
Fixes #1729.

### CODE REVIEW

**Comment by jonas054** [comment]
A couple of lines a too long. Other than that, it looks good.

I guess we have a lot of places where we can add calls to `unexpected_style_detected` now. :smile: For some reason it never occurred to me that we can have `--auto-gen-config` generate configuration for a consistently used non-default style when there are more than two choices. Nice!

File: lib/rubocop/ast_node.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+module Astrolabe
+  # RuboCop's extensions to Astrolabe::Node (which extends Parser::AST::Node)
+  #
+  # Contribute as much of this as possible to the `astrolabe` gem
+  # If any of it is accepted, it can be deleted from here
+  #
+  class Node
+    def multiline?
+      expr = loc.expression
+      expr && expr.source.include?("\n")
```
Comment (rrosenblum) [inline lib/rubocop/ast_node.rb]: I do not think this is accurate at determining if all types of nodes are multiline. Strings being the main issue as they could include `"\n"` within the string while still being on one line.

Something similar came up in a recent PR. If I remember correctly, the code used was
```ruby
node.loc.expression.first_line != node.loc.expression.last_line
```

Comment (alexdowad) [inline lib/rubocop/ast_node.rb]: The issue with strings sounds very doubtful. If you have a string like `"a line\nanother line"`, then the source should include `'\n'` but not `"\n"`, if you know what I mean. But the suggested code looks like it should be much more efficient on large nodes. I'm updating it now... thanks.

**Comment by alexdowad** [comment]
Updated according to feedback from @jonas054 and @rrosenblum. Thanks to both of you!

**Comment by bbatsov** [comment]
:+1: Nice cleanup!

**Pull request merged**